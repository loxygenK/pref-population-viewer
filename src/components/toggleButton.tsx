import styles from "./toggleButton.module.scss";

export interface ToggleButtonProps {
  className?: string;
  captionWhenUnpressed: string;
  captionWhenPressed: string;
  pressed: boolean;
  onChange: (newPressedState: boolean) => void;
}
export const ToggleButton: React.FC<ToggleButtonProps> = ({
  className,
  captionWhenPressed,
  captionWhenUnpressed,
  pressed,
  onChange,
}) => {
  const onButtonClicked = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();

    onChange(!pressed);
  };

  return (
    <button
      className={`${styles.toggle_button} ${pressed ? styles.pressed : ""} ${
        className ?? ""
      }`}
      onClick={onButtonClicked}
    >
      {pressed ? captionWhenPressed : captionWhenUnpressed}
    </button>
  );
};
