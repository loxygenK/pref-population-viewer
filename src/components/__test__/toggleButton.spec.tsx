import renderer from "react-test-renderer";
import { ToggleButton } from "../toggleButton";

describe("Section wrapper component", () => {
  it("renders correctly when button is not pressed", () => {
    const tree = renderer
      .create(
        <ToggleButton
          captionWhenUnpressed="Unpressed"
          captionWhenPressed="Pressed"
          pressed={false}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when button is pressed", () => {
    const tree = renderer
      .create(
        <ToggleButton
          captionWhenUnpressed="Unpressed"
          captionWhenPressed="Pressed"
          pressed={true}
          onChange={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
