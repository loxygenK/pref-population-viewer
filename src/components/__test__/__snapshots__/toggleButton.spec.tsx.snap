// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section wrapper component renders correctly when button is not pressed 1`] = `
<button
  className="toggle_button  "
  onClick={[Function]}
>
  Unpressed
</button>
`;

exports[`Section wrapper component renders correctly when button is pressed 1`] = `
<button
  className="toggle_button pressed "
  onClick={[Function]}
>
  Pressed
</button>
`;
